pkg/virt-controller/watch/vmi.go

```diff
@@ -380,6 +380,40 @@ func (c *VMIController) updateStatus(vmi *virtv1.VirtualMachineInstance, pod *k8
 			conditionManager.RemoveCondition(vmiCopy, virtv1.VirtualMachineInstanceConditionType(k8sv1.PodReady))
 		}
 
+		var networkPatchNeeded bool
+		existingNetworks := map[string]virtv1.Network{}
+		for _, network := range vmiCopy.Spec.Networks {
+			if network.Name != "" {
+				existingNetworks[network.Name] = network
+			}
+		}
+		for i, iface := range vmiCopy.Status.Interfaces {
+			if existingNetworks[iface.Name].NetworkSource.Pod != nil && vmiCopy.Status.Interfaces[i].IP != pod.Status.PodIP {
+				vmiCopy.Status.Interfaces[i].IP = pod.Status.PodIP
+				networkPatchNeeded = true
+
+				for j, ip := range iface.IPs {
+					if ip == iface.IP {
+						vmiCopy.Status.Interfaces[i].IPs[j] = pod.Status.PodIP
```

Comment:
Since we report only the PodIP in IPs with masquerade, we should not really need to iterate all the IPs. It should be sufficient to only se the IPs to [IP]. Am I missing some scenario when this is not the case?